kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/SequenceEq.kt

```diff
@@ -0,0 +1,32 @@
+package io.kotest.assertions.eq
+
+import io.kotest.assertions.Actual
+import io.kotest.assertions.Expected
+import io.kotest.assertions.failure
+import io.kotest.assertions.print.print
+
+object SequenceEq : Eq<Sequence<*>> {
+
+   /**
+    * [Sequence] is not a supported type.  Sequences can be infinite, and should be
+    * compared only with custom code.
+    */
+
+   fun isValidSequence(it: Any): Boolean {
+      return when (it) {
+         is Sequence<*> -> true
+         else -> false
+      }
+   }
+
+   fun asSequence(it: Any): Sequence<*> {
+      return when (it) {
+         is Sequence<*> -> it
+         else -> error("Cannot convert $it to Sequence<*>")
+      }
+   }
+
+   override fun equals(actual: Sequence<*>, expected: Sequence<*>, strictNumberEq: Boolean): Throwable? =
```

Comment:
Since anyway we are failing here, not sure then we even need this is eq.

Reply:
Included for symmetry, and to push an error message that is explicit about the reason for failing. No strong feelings :)  Unconditional failure also is easy to revise if anything changes, but this latter is a weaker justification.